### ISSUE TITLE
dev: Ledger "Claiming GAS failed"

### ISSUE THREAD

**Issue by brianlenz** [CityOfZion/neon-wallet#833]
I've been able to successfully claim GAS on a TestNet wallet on my Ledger, but I've now tried 4 times to claim GAS on my MainNet wallet (including restarting Neon), and it's failed every time with:

> Claiming GAS failed.

I sign the send transaction, and it goes through successfully. Then, I'm properly prompted to sign the claim transaction. Immediately after signing the claim tx on the Ledger, the error message displays.

I don't know if it's something about the wallet differences, but I can claim GAS successfully every time on my TestNet Ledger wallet (which is a Ledger wallet without a passphrase). My MainNet wallet that I'm claiming with is a different Ledger wallet (same device, but it has a passphrase and thus different PIN), and it fails every time. I don't see why that would make a difference, but I thought I'd mention it in case it's related.

**Comment by mhuggins** [comment]
Based upon your experience, it sounds like an issue with the neon-js `api.claimGas` call itself, which is the third and last step in the claim process.

cc @snowypowers in case he has any thoughts. (I'm not in front of a computer all day, sorry.)

**Comment by snowypowers** [comment]
My experience with this is usually neoscan/neonDB is returning the wrong data, and thus our claim transaction fails. A good way to check this is to query both neoscan and neonDB with your address and compare the data. 
Currently, neoscan has a higher rate of failure compared to neonDB but this would not push the switch towards neonDB as this is not considered an API failure

**Comment by ducaalfa** [comment]
something similar happens to me using Ledger Nano s
when i try to send gas to another neon wallet, neon asks to confirm, disappears in seconds (couldn't confirm) says Error: Transaction failed. No way to confirm TX, error arrives to fast before.
Nano firmware is 1.4.1
Tried to uninstall neon and uninstalled ledger neo software, reinstall them all but no changes
Neo transfert works great instead
Anyone can help..

**Comment by mhuggins** [comment]
Sounds like this will require looking into a couple issues:

1. Both neonDB/neoscan may calculate unclaimed GAS incorrectly in some circumstances.
2. For Ledger devices, if the transaction is too large (e.g.: vin is too large because of lots of small GAS claims), then it's too large for Ledger to handle.

Neither of these are new issues and IMO shouldn't block a release.

**Comment by ducaalfa** [comment]
hi dmlipat 
I think you were lucky, I reverted too to 0.2.1 and 0.4.1 older versions but no changes.
Here you have a more complete description of my situation:
I own a nano s fully updated (security element 1.4.1 and MCU 1.5) and (now) last Neon wallet installed (0.2.2).
It is 2 weeks that I do not succeed in sending GAS tokens to other neo wallets since there is no way to confirm TX, error transaction fail arrives too fast before i can push button
This happens only with gas, if i send neo coins everything goes fine.
This issue happens not only on my PC but also using different PCs and it happens also using older Neon versions.
I tried to reinstall all Ledger apps on PC, I tried to reinstall Neon Wallet, I tried also nano reset but no way out, every time still the same transaction fail.
What to do now ? I bought Ledger to increase security but I am realizing I have stuck coins in it without any opportunity to arrange my funds...
thanks for help!

**Comment by dmlipat** [comment]
ducaalfa
i can feel you frustration. try this, uninstall then delete the neon folder under appdata/roaming? if possible clean registry by using ccleaner. then install the old version.

**Comment by GoldVintage** [comment]
Hi all, 

Exactly the same issue...

thnaks for your help!

**Comment by lavalleynh** [comment]
Still an issue in v0.2.4, 100% fail to calculate claimable gas.

**Comment by fahadysf** [comment]
Issue is still present in latest version of NEON wallet.

**Comment by prouast** [comment]
Can confirm, I have the same issue with v0.2.4 and latest Ledger firmware.

**Comment by mrbrad11** [comment]
I had the same problem with version 0.2.5 too. I was finally able to claim GAS when I turned disconnected and reconnected my internet connection to the computer. Try turning WiFi on and off or unplugging and plugging back in your ethernet!

**Comment by bitoutshop** [comment]
Using my Ledger nano S. I have 2 separate Ledger’s both of which worked the last time I used them a few months ago.
•	Claiming GAS = Error calculating Gas: Transaction failed
 

•	Attempting to Send 1 NEO to a Different Address = Transaction Failed: Timeout of 5000ms Exceeded
 


•	Set my Computer to be completely open to the internet, NAT all traffic too and from my PC directly out the Internet connection
•	Attempt to Send 1 NEO to a different address. = Transaction Failed : Transaction Failed
 

I have tried
1.	Restarting Router and PC
2.	Unplugging and plugging in my Ledger to different USB port
3.	A completely different PC on a different network.


Cant see anything in the neon/logs Directory.
How can I troubleshoot this?
